lab_2_retrieval_w_bm25/main.py

```diff
@@ -77,6 +138,31 @@ def calculate_idf(vocab: list[str], documents: list[list[str]]) -> dict[str, flo
 
     In case of corrupt input arguments, None is returned.
     """
+    if (not isinstance(vocab, list)
+            or not all(isinstance(uniq_word, str) for uniq_word in vocab)
+            or not vocab):
+        return None
+    if (not isinstance(documents, list)
+            or not all(isinstance(one_doc_text, list) for one_doc_text in documents)
+            or not documents):
+        return None
+    for one_doc_text in documents:
+        if (not all(isinstance(word, str) for word in one_doc_text)
+                or not one_doc_text):
+            return None
+
+    idf_figure_dict = {}
+    for word in vocab:
+        for one_doc_text in documents:
```

Comment:
Do you really need to duplicate this condition? It's already on the next line

Reply:
![image](https://github.com/user-attachments/assets/ed1812ee-8487-4f94-90dc-803bb87a6fac)
this variable will not work(